### ISSUE TITLE
KeepKey v7.1.0 not being detected

### ISSUE THREAD

**Issue by NTICompass** [greatwolf/KKClient#4]
I upgraded my KeepKey to firmware 7.1.0 (on bootloader 2.0.0) and KKClient is no longer detecting the device.

Here is what I see in Chrome's console:

```
Keepkey Chrome Application Wallet: 6.6.4
Background.js:332 timer done
Background.js:259 looking for HID devices...
Background.js:269 []
Background.js:279 looking for WebUsb devices...
Background.js:287 [{"device":0,"manufacturerName":"KeyHodlers, LLC","productId":2,"productName":"KeepKey - NTICompass Inc.","serialNumber":"<serial goes here>","vendorId":11044,"version":256}]
Background.js:289 Initializing WebUsb KeepKey
Background.js:292 dispatching
Background.js:9626 WebUsb connecting Object
Background.js:9431 WebUsb openDevice Object
Background.js:9439 WebUsb setConfiguration Object handle: {handle: 1, productId: 2, vendorId: 11044}
Background.js:9445 WebUsb setConfiguration failed
Background.js:9446 Error setting device configuration.
```

Then later on, I see:

```
sending message
Background.js:13406 device --> proxy: [Features]
 {
    "vendor": "keepkey.com",
    "major_version": 7,
    "minor_version": 1,
    "patch_version": 0,
    "bootloader_mode": null,
    "device_id": "<my device id>",
    "pin_protection": true,
    "passphrase_protection": false,
    "language": "english",
    "label": "NTICompass Inc.",
    "coins": [],
    "initialized": true,
    "revision": "64306231346639333963346166663838306237383364303437616436396238643631623036616638",
    "bootloader_hash": "9bf1580d1b21250f922b68794cdadd6c8e166ae5b15ce160a42f8c44a2f05936",
    "imported": false,
    "pin_cached": false,
    "passphrase_cached": false,
    "policies": [
        {
            "policy_name": "ShapeShift",
            "enabled": true
        },
        {
            "policy_name": "Pin Caching",
            "enabled": true
        },
        {
            "policy_name": "Experimental",
            "enabled": false
        },
        {
            "policy_name": "AdvancedMode",
            "enabled": false
        }
    ],
    "model": "K1-14AM",
    "firmware_variant": "KeepKey",
    "firmware_hash": "d8b2b43eada45ded399f347289750a7083081186b37158b85eab41a38cbc6e50",
    "no_backup": false,
    "typeName": "Features"
}
Background.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
```

This is working on a previous firmware version, but not this one.

**Comment by greatwolf** [comment]
Yeah this is a known limitation with how the extension checks the firmware version during startup.

Basically what happens is it tries to match the firmware on the device with a list of known firmware profiles(search for `155:` in background.js to find where it's defined). The extension will stop loading if it cannot find a matching version in its list.

Thankfully most of the profile attributes don't really change between firmware versions. This means you can copy/paste an existing profile and bump the version fields to 7.1.0 and it should load again.

Another way you can fix it is to add this to the end of the profile array:

    {
      identity:
      {
        vendor: "keepkey.com",
        major_version: 7,
        bootloader_mode: null
      },
      capabilities:
      {
        vendorName: "KeepKey",
        firmwareImageAvailable: !0,
        defaultMnemonicSeedLength: 12,
        supportsCipheredKeyRecovery: !0,
        supportsSecureAccountTransfer: !0,
        usesProtectCallButtonRequests: !1,
        usesShapeshiftResponseV2: !0,
        upgradeSkipable: !0,
        hasPagedCoinTable: !0,
        supportsRecoveryDryRun: !0
      }
    },

This profile will actually match successfully with *7.x.x* firmware which will allow loading any version as long as the major version is `7`. This will provide some future-proofing for later versions.

I have a commit that fixes this issue 305c62b in the master branch which will be in the next release.

**Comment by greatwolf** [comment]
Closing this since 6.6.5 just released fixes this.
